cmd/zt_copy_file_file_test.go

```diff
@@ -0,0 +1,251 @@
+
+	// all files at source should be copied to destination
+	runCopyAndVerify(c, raw, func(err error) {
+		c.Assert(err, chk.IsNil)
+
+		// validate that the right number of transfers were scheduled
+		c.Assert(len(mockedRPC.transfers), chk.Equals, len(fileList))
+
+		// validate that the right transfers were sent
+		validateS2STransfersAreScheduled(c, "/", "/", fileList, mockedRPC)
+	})
+
+	// turn off recursive, we should be getting an error
+	raw.recursive = false
+	mockedRPC.reset()
+	runCopyAndVerify(c, raw, func(err error) {
+		c.Assert(err, chk.NotNil)
+		// make sure the failure was due to the recursive flag
+		c.Assert(strings.Contains(err.Error(), "recursive"), chk.Equals, true)
+	})
+}
+
+// include flag limits the scope of source/destination comparison
+func (s *cmdIntegrationSuite) TestFileCopyS2SWithIncludeFlag(c *chk.C) {
+	fsu := getFSU()
+	srcShareURL, srcShareName := createNewAzureShare(c, fsu)
+	dstShareURL, dstShareName := createNewAzureShare(c, fsu)
+	defer deleteShare(c, srcShareURL)
+	defer deleteShare(c, dstShareURL)
+
+	// set up the source share with numerous files
+	fileList := scenarioHelper{}.generateCommonRemoteScenarioForAzureFile(c, srcShareURL, "")
+	c.Assert(len(fileList), chk.Not(chk.Equals), 0)
+
+	// add special files that we wish to include
+	filesToInclude := []string{"important.pdf", "includeSub/amazing.jpeg", "exactName"}
+	scenarioHelper{}.generateAzureFilesFromList(c, srcShareURL, filesToInclude)
+	includeString := "*.pdf;*.jpeg;exactName"
+
+	// set up interceptor
+	mockedRPC := interceptor{}
+	Rpc = mockedRPC.intercept
+	mockedRPC.init()
+
+	// construct the raw input to simulate user input
+	srcShareURLWithSAS := scenarioHelper{}.getRawShareURLWithSAS(c, srcShareName)
+	dstShareURLWithSAS := scenarioHelper{}.getRawShareURLWithSAS(c, dstShareName)
+	raw := getDefaultCopyRawInput(srcShareURLWithSAS.String(), dstShareURLWithSAS.String())
+	raw.include = includeString
+	raw.recursive = true
+
+	// verify that only the files specified by the include flag are copyed
+	runCopyAndVerify(c, raw, func(err error) {
+		c.Assert(err, chk.IsNil)
+		validateS2STransfersAreScheduled(c, "/", "/", filesToInclude, mockedRPC)
+	})
+}
+
+// exclude flag limits the scope of source/destination comparison
+func (s *cmdIntegrationSuite) TestFileCopyS2SWithExcludeFlag(c *chk.C) {
+	fsu := getFSU()
+	srcShareURL, srcShareName := createNewAzureShare(c, fsu)
+	dstShareURL, dstShareName := createNewAzureShare(c, fsu)
+	defer deleteShare(c, srcShareURL)
+	defer deleteShare(c, dstShareURL)
+
+	// set up the source share with numerous files
+	fileList := scenarioHelper{}.generateCommonRemoteScenarioForAzureFile(c, srcShareURL, "")
+	c.Assert(len(fileList), chk.Not(chk.Equals), 0)
+
+	// add special files that we wish to exclude
+	filesToExclude := []string{"notGood.pdf", "excludeSub/lame.jpeg", "exactName"}
+	scenarioHelper{}.generateAzureFilesFromList(c, srcShareURL, filesToExclude)
+	excludeString := "*.pdf;*.jpeg;exactName"
```

Comment:
Ensure that generateCommonRemoteScenarioForAzureFile() doesn't end up generating *.pdf or *.jpeg as we don't do any manual filtering of the file list in this test

Reply:
Yes, thanks for the reminder